react/ProductSummary.js

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Button from '@vtex/styleguide/lib/Button'
+import { FormattedMessage } from 'react-intl'
+
+import ProductName from './ProductName'
+import Price from './Price'
+import DiscountBadge from './DiscountBadge'
+
+/**
+ * Product Summary component. Summarizes the product informations.
+ */
+class ProductSummary extends Component {
+  /**
+   * Constructs the state of the component. Uses a hovering property to react if the Buy button should be shown only on hover.
+   *
+   * @param {*} props Component's props
+   */
+  constructor(props) {
+    super(props)
+    this.state = { isHovering: false }
+  }
+
+  onMouseLeave() {
+    this.setState({ isHovering: false })
+  }
+
+  onMouseEnter() {
+    this.setState({ isHovering: true })
+  }
+
+  /**
+   * Act like an anchor tag, redirecting to the product URL or opening a new tab if the ctrl key is pressed.
+   *
+   * @param {Event} event Click event
+   */
+  redirect(event) {
+    event.ctrlKey ? window.open(this.props.product.url) : window.location.assign(this.props.product.url)
+  }
+
+  render() {
+    const {
+      product,
+      showListPrice,
+      showLabels,
+      showInstallments,
+      showBadge,
+      badgeText,
+      hideBuyButton,
+      showButtonOnHover,
+    } = this.props
+
+    return (
+      <div className="tc pointer"
+        onMouseEnter={() => this.onMouseEnter()}
+        onMouseLeave={() => this.onMouseLeave()}>
+        {!product &&
+          <div>
+            <FormattedMessage id="loading" />
+          </div>
+        }
+        {product && (
+          <div>
+            <div onClick={event => this.redirect(event)}>
```

Comment:
Idem

Reply:
Chama `this.props.product.url`

Reply:
Dá para usar props também se você fizer o bind com arrow functions.